the-collab-lab/tcl-26-smart-shopping-list
# Item details accordion

## Description

This PR adds an accordion style display to show item details inline with the list items. There is a show/hide all details button in the header allowing the user more control over how much information they want to see, as well as a toggle button for each individual item. This is done with a state showAllDetails within ListView, a state showSingleDetail within each item, and a useEffect callback within each item that updates its own local showSingleDetail state when showAllDetails is updated.

We've also used Luxon to format the dates to provide item details, and added labels to items needing to be purchased soon, and a message notifying the user that an item does not seem to be being bought anymore for inactive items with a suggestion to delete.

## Related Issue

Closes #57
Closes #84
Closes #82

## Acceptance Criteria

- [x] Item details area is shown/hidden when user clicks the details icon
- [x] Include proper aria attributes
- [x] Details for the items are printed from the database, including formatted dates
- [x] Date only shows year if not the current year (to save space)
- [x] Details arrow icon points up when the list is expanded (there's already a class for this called `item__details-button_expanded`)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### After

![image](https://user-images.githubusercontent.com/67769490/131760302-5ee1e409-3e78-495d-bf47-94495e86f837.png)


## Testing Steps / QA Criteria

1. Pull down the branch and run npm start.
2. Check to make sure the dates are formatted, and the labels are printed on the correct items.
3. Check to make sure the 'show/hide all details' link at the top works, as well as the individual toggle buttons on each item.
